sycl/source/detail/event_impl.cpp

```diff
@@ -250,16 +251,25 @@ void event_impl::cleanupCommand(
     detail::Scheduler::getInstance().cleanupFinishedCommands(std::move(Self));
 }
 
+void event_impl::checkProfilingPreconditions() const {
+  QueueImplPtr QueueImpl = MQueue.lock();
+  if (QueueImpl &&
+      !QueueImpl->has_property<property::queue::enable_profiling>()) {
```

Comment:
Now that you have `MIsProfilingEnabled` it would be better to use it here to prevent a case where the queue dies before the event.